Fix turtle mode motor direction on rearm

haslinghuis opened the pull request:
- fixes https://github.com/betaflight/betaflight/pull/13905#issuecomment-2476842535
- alternative to #14358

ledvinap commented:
keeping motors reversed when disarmed may cause problems (configutator motor test, idle motors during disarm). IMO or is safer to set direction two times

haslinghuis commented:
`#ifdef USE_DSHOT` starts at 534 and ends at 583

nerdCopter reviewed (approved):
- pre-approving for workflow. requested @ cheetos testing.

ledvinap commented on src/main/fc/core.c:
```diff
@@ -570,13 +573,11 @@ void tryArm(void)
                     setMotorSpinDirection(DSHOT_CMD_SPIN_DIRECTION_REVERSED);
                 }
-#endif
             }
```
you can set it only in one place  (removing setMotorSpinDirection above):
```suggestion
            }
            setMotorSpinDirection(crashFlipModeActive ? DSHOT_CMD_SPIN_DIRECTION_REVERSED : DSHOT_CMD_SPIN_DIRECTION_NORMAL);
```
```

nerdCopter commented:
user `@ Cheetos` feedback:
```
this still works great
✅  guuud fix
Support data submitted   Id: "55ade196-07e1-4b1e-abb4-ea1c7f446ca4"
```

ledvinap reviewed (approved):
It is only workaround, but it should mask some direction-related bugs

ctzsnooze reviewed (approved):
Thanks very much guys

SteveCEvans commented:
I don't see how this fixes anything. It simply prevents `DSHOT_CMD_SPIN_DIRECTION_NORMAL` being sent when not in turtle mode, but doesn't address the commands not being sent at all.

haslinghuis commented:
This pull request refactors the logic for setting motor spin direction during the arming process in the `tryArm` function of `src/main/fc/core.c`. The changes simplify the code by consolidating conditional logic and removing redundant preprocessor directives.

### Refactoring of motor spin direction logic:

* Removed the `#ifdef USE_DSHOT` block that conditionally set motor spin direction based on crash flip mode. The logic is now simplified and always applies, with the motor spin direction set using a single line of code. (`[src/main/fc/core.cL573-R574](diffhunk://#diff-b73d61d161176fe900ec8e26672ec0c91a08f621ee5185caadb44de19f139cffL573-R574)`)

The change did not take into account underlying factors as there were no reports so far.

nerdCopter commented:
so should this be reverted or no?
